core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -81,7 +81,7 @@
 
   /** Executor for application processing. */
   private Executor executor;
-  @GuardedBy("lock") private boolean usingSharedExecutor;
+  private boolean usingSharedExecutor;
```

Comment:
I don't feel comfortable removing documentation that describes how variables should be properly synchronized.  This needs a comment (compiled as annotation or otherwise) describing how to safely access executor, usingSharedExecutor, and the other variables in this class.  

I don't think moving usingSharedExecutor and executor to inside the synchronized block is overkill.  What bad thing is going to happen?